packages/pcd-types/src/pcd.ts

```diff
@@ -265,3 +265,20 @@ export type PCDArgument<T extends PCD = PCD> = Argument<
 export function isPCDArgument(arg: Argument<any, unknown>): arg is PCDArgument {
   return arg.argumentType === ArgumentTypeName.PCD;
 }
+
+export type RevealMap = Record<string, boolean>;
+export type RevealMapArgument<T extends RevealMap> = Argument<
+  ArgumentTypeName.Object,
+  T
+>;
+export function isRevealMapArgument(
```

Comment:
i think adding an explicit new type here, and via `pcd.ts` makes sense. That way it could also show up as a 'Toggle List' in the UI, rather than an 'Object'